integration-tests/test-cases/scenario-tests/src/test/java/io/cellery/integration/scenario/tests/EmployeePortalTestCase.java

```diff
@@ -35,56 +48,89 @@
     private static final String VERSION = "1.0.0";
     private static final String LINK_STOCK = "stockCellDep:stock-inst";
     private static final String LINK_EMPLOYEE = "employeeCellDep:employee-inst";
+    private HttpClient httpClient;
 
-    @Test
+    @BeforeClass
+    public void setup() {
+        this.httpClient = new HttpClient();
+    }
+
+    @Test(description = "Tests the building of stock cell image.")
     public void buildStock() throws Exception {
         build("stocks.bal", Constants.TEST_CELL_ORG_NAME, STOCK_IMAGE_NAME, VERSION,
                 Paths.get(CELLERY_SCENARIO_TEST_ROOT, "employee-portal", "stock").toFile().getAbsolutePath());
     }
 
-    @Test
+    @Test(description = "Tests the running of stock cell instance.")
```

Comment:
Do we need separate test cases for starting each cells? I think you simply can use the start with dependencies option here to start. Because the starting each cells individually, is already addressed in other scenario cases. Probably, you should start only stock cell instance, and start directly the HR cell with --start-dependencies param along with -l params, and hence you can make sure the remaining the employee cell is started.

Reply:
Updated as follows.
1. One test case to build all cell images (stock, employee, hr). Note that sample cells are not pushed to docker hub yet.
2. One test case to run all the tests.
    * Run stock cell independently.
    * Run hr cell by giving link to running stock cell instance and starting dependent employee cell instance.